src/python/pants/build_graph/build_graph.py

```diff
@@ -520,6 +520,13 @@ def resolve_address(self, address):
     :returns: The Addressable which address points to.
     """
 
+  @abstractmethod
+  def fork(self):
+    """Fork a buildgraph from scratch without copying its state.
```

Comment:
The terminology "fork" is unclear. Usually I would take that to mean something like "clone". Can you elaborate in the docstring what is copied and what is not?